### DISCUSSION TITLE
May I fix this virtualization bug?

### DISCUSSION THREAD

**Discussion by DavidRieman** [necolas/react-native-web#2236] [Q&A] [answered]
[Over here](https://github.com/necolas/react-native-web/issues/1608#issuecomment-1043625089) I alluded to an RNW bug that I've since dug more into and it seems to be a general virtualization bug and not necessarily just FlatList.  I've tried updating a few files in the vendor area (FlatList, VirtualizeUtils, and VirtualizedList) from the latest at https://github.com/facebook/react-native.  That seems to make some severe virtualization bugs go away!  🎉 

I'd like to know if I putting up a PR for this would be likely to receive timely attention and a decent reception, or whether there'd be reasons to stop such an effort...  So hoping to hear something before putting together the PR.  (My main alternative, of course, is to move our product to referencing a fork.)  Since I have a couple tiny PRs already up awaiting feedback, I'd also like to see them through the process first as a trial run before ramping up to a PR with actual substance to it.

Thank you again for maintaining such a wonderful library.  😄

**Comment by necolas** [comment, marked as answer]
Yes PR is welcome to update the vendor files

**Comment by DavidRieman** [comment]
My approach so far - I made some bite sized PRs (3 up right now #2229, #2237, #2238) with the bits leading up to easily reproducing FlatList-flavored VirtualizedList issues directly through the `examples` project.  I'll continue targeting individual feature merges to this base repo, but will also be merging everything to my own master branch to have the latest with all those parts up and running, so we can easily modify some `examples` page to test things out.  None of those parts are _critical_ to the virtualized list issues themselves but I want them in my own main branch to work with daily as I get these things updated.

Right now I've got _local edits_ which took a manual diff of the latest code from [facebook/react-native](https://github.com/facebook/react-native) to update react-native-web's `src/vendor/react-native` copies from.  I took all changes that I could to FlatList, VirtualizeUtils, and VirtualizedList (but manually adjusted the imports).  There will be one debug-only feature that I'm going to skip (and log a TODO ticket for later) as a ScrollView feature is not present.  I haven't run the tests yet.  The good news is that the big list virtualization issues I was seeing seem to be fixed by taking this fresh parity with `react-native` itself.

Next things on my TODO list here are:
[x] Get a PR up with an `examples` page for demonstrating basic `SectionList` support (because that's also a virtualized list).
[x] Update SectionList vendor files from latest RN code as well.
[x] Look for any other VirtualizedList-related things that might also need to update in tandem.
[x] Rename `master` to `main` at least for my repo.
[x] Test that these virtualization fixes actually solve the problems we're having in our actual application environment. (Update: Things are improved but not fully solved _for our app's complex scenario_ by this round of changes.  Still hoping these PRs move through to fix parts that are fixed.)
[x] Get all the PRs up.
[ ] Get all the PRs through.  **BLOCKED** awaiting maintainer feedback / validation/ acceptance.
